rapid7/metasploit-framework
# Fix crash when generating payload sizes

Fixes a crash when generating payload sizes

### Before

Adaptor payloads are updated in size, and other modules crash:

```
$ ./tools/modules/update_payload_cached_sizes.rb
[*] Updating the CacheSize for /Users/adfoster/Documents/code/metasploit-framework/modules/payloads/adapters/cmd/unix/python.rb...
[*] Updating the CacheSize for /Users/adfoster/Documents/code/metasploit-framework/modules/payloads/adapters/cmd/unix/python.rb...
[*] Updating the CacheSize for /Users/adfoster/Documents/code/metasploit-framework/modules/payloads/adapters/cmd/unix/python.rb...
.... etc ...
[-] Caught Error while updating cmd/windows/powershell/custom/reverse_hop_http:
wrong number of arguments (given 1, expected 0)
[-] Caught Error while updating cmd/windows/powershell/custom/reverse_http_proxy_pstore:
wrong number of arguments (given 1, expected 0)
[-] Caught Error while updating cmd/windows/powershell/custom/reverse_https_proxy:
wrong number of arguments (given 1, expected 0)
[-] Caught Error while updating cmd/windows/powershell/dllinject/reverse_hop_http:
wrong number of arguments (given 1, expected 0)
```

The errors were being generated by payloads which used a method definition of `def generate(opts = {})` which called `super` with a parent class of `def generate`

### After

No crashes, and the adaptor payload's size is no longer updated multiple times

## Verification

- Verify no more crashes
- Verify the payload metadata is ignored for adaptors
